storage/xtradb/ut/ut0ut.cc

```diff
@@ -395,21 +398,25 @@ Runs an idle loop on CPU. The argument gives the desired delay
 in microseconds on 100 MHz Pentium + Visual C++.
 @return	dummy value */
 UNIV_INTERN
-void
+ulint
 ut_delay(
 /*=====*/
 	ulint	delay)	/*!< in: delay in microseconds on 100 MHz Pentium */
 {
-	ulint	i;
+	ulint	i, j;
 
-	UT_LOW_PRIORITY_CPU();
+	j = 0;
 
 	for (i = 0; i < delay * 50; i++) {
+		j += i;
 		UT_RELAX_CPU();
-		UT_COMPILER_BARRIER();
 	}
 
-	UT_RESUME_PRIORITY_CPU();
+	if (ut_always_false) {
+		ut_always_false = (ibool) j;
+	}
+
+	return(j);
 }
 #endif /* !UNIV_HOTBACKUP */
 
```

Comment:
This is not good idea, if there is a need for different ut_delay for AIX, provide a new function instead.